File: Framework/Qt/Python/Chapter_01/QT01_04.py
"""
Setup Qt application software containing widgets in a main window.
"""

from PySide2.QtWidgets import QApplication, QMainWindow, QLabel

if __name__ == '__main__':
    app = QApplication()
    window = QMainWindow()

    # DECLARE label widget with "Hello World!"
    label = QLabel("Hello World!")

    '''
    IMPORTANT: Widgets are displayed through the CentralWidget in Windows.
    '''

    # SET label widget as the Central Widget of the MainWindow.
    window.setCentralWidget(label)

    window.show()
    app.exec_()

File: Framework/Qt/Cpp/Chapter_01/QT01_04/QT01_04.cpp
/*
Setup Qt application software containing widgets in a main window.
*/

#include <qapplication.h>
#include <qmainwindow.h>
#include <qlabel.h>

int main(int argc, char **argv) {
	QApplication app(argc, argv);
	QMainWindow window;

	// DECLARE label widget with "Hello World!"; pointer to a QLabel in heap memory.
	QLabel *label = new QLabel("Hello World!");

	/*
	IMPORTANT: Widgets are displayed through the CentralWidget in Windows.
	*/

	// SET label widget as the Central Widget of the MainWindow; accepts pointer as argument.
	window.setCentralWidget(label);

	window.show();
	return app.exec();
}